apps/client/src/features/app-settings/panel/network-panel/GenerateLinkForm.tsx

```diff
@@ -0,0 +1,136 @@
+import { useState } from 'react';
+import { useForm } from 'react-hook-form';
+import QRCode from 'react-qr-code';
+import { Button, Select, Switch } from '@chakra-ui/react';
+
+import { generateUrl } from '../../../../common/api/session';
+import { maybeAxiosError } from '../../../../common/api/utils';
+import ExternalLink from '../../../../common/components/external-link/ExternalLink';
+import Info from '../../../../common/components/info/Info';
+import useInfo from '../../../../common/hooks-query/useInfo';
+import useUrlPresets from '../../../../common/hooks-query/useUrlPresets';
+import copyToClipboard from '../../../../common/utils/copyToClipboard';
+import { preventEscape } from '../../../../common/utils/keyEvent';
+import { linkToOtherHost } from '../../../../common/utils/linkUtils';
+import { serverURL } from '../../../../externals';
+import * as Panel from '../../panel-utils/PanelUtils';
+
+import style from './GenerateLinkForm.module.scss';
+
+interface GenerateLinkFormOptions {
+  baseUrl: string;
+  path: string;
+  lock: boolean;
+  authenticate: boolean;
+}
+
+type GenerateLinkState = 'pending' | 'loading' | 'success' | 'error';
+
+export default function GenerateLinkForm() {
+  const { data: infoData } = useInfo();
+  const { data: urlPresetData } = useUrlPresets();
+  const [formState, setFormState] = useState<GenerateLinkState>('pending');
+  const [url, setUrl] = useState(serverURL);
+
+  const {
+    handleSubmit,
+    register,
+    setError,
+    formState: { errors },
+  } = useForm<GenerateLinkFormOptions>({
+    mode: 'onChange',
+    resetOptions: {
+      keepDirtyValues: true,
+    },
+  });
+
+  const onSubmit = async (options: GenerateLinkFormOptions) => {
+    try {
+      setFormState('loading');
+      const baseUrl = linkToOtherHost(options.baseUrl);
```

Comment:
dose this work past the proxy?
eg will it get the port proxy port or internal ontime port?

Reply:
can you please double-check whether the tests cover the possible cases?